bundle/QueryType/SearchQueryType.php

```diff
@@ -27,38 +31,77 @@ public static function getName(): string
 
     protected function configureOptions(OptionsResolver $optionsResolver): void
     {
-        $optionsResolver->setRequired(['search_text', 'content_types', 'subtree']);
-
+        $optionsResolver->setRequired(['search_text', 'content_types', 'subtree', 'sort', 'order']);
         $optionsResolver->setAllowedTypes('search_text', 'string');
         $optionsResolver->setAllowedTypes('content_types', 'string[]');
+        $optionsResolver->setAllowedTypes('sort', 'string[]');
         $optionsResolver->setAllowedTypes('subtree', ['int', 'string']);
-
         $optionsResolver->setAllowedValues(
             'search_text',
             static fn (string $searchText): bool => trim($searchText) !== '',
         );
-
+        $optionsResolver->setAllowedValues(
+            'sort',
+            /** @var string[] $keys */
+            fn (array $keys): bool => $this->sortKeysAllowed($keys),
+        );
+        $optionsResolver->setAllowedValues(
+            'order',
+            static fn (string $searchText): bool => trim($searchText) === 'asc' || trim($searchText) === 'desc',
```

Comment:
No need for `trim` and we should use `Query::SORT_*` constants for easier handling later on.

You can also just say `in_array($order, [Query::SORT_ASC, Query::SORT_DESC], true)`.

Also why `$searchText`? :D

Reply:
Fixed!